peps/pep-0763.rst

```diff
@@ -0,0 +1,275 @@
+
+* April 2023: `codecov <https://pypi.org/project/codecov/>`_ was deleted by
+  its maintainers after a long deprecation period. This caused extensive
+  breakage for many of Codecov's CI/CD users, who were unaware of the
+  deprecation period due to limited observability of deprecation warnings
+  within CI/CD logs.
+
+  The project was
+  `subsequently re-created <https://about.codecov.io/blog/message-regarding-the-pypi-package/>`_
+  by its maintainers, with a new release published to compensate for the deleted releases
+  (which were not restored), meaning that any pinned installations remained
+  broken. As of October 2024, this single release remains the only release on
+  PyPI and has around
+  `1.5 million monthly downloads <https://pypistats.org/packages/codecov>`_.
+
+* June 2023: `python-sonarqube-api <https://pypi.org/project/python-sonarqube-api/>`_
+  deleted all released versions prior to 2.0.2, which was both a breaking
+  change version *and* was accompanied by a paid "Professional Edition".
+
+  The project's maintainer subsequently
+  `deleted conversations <https://discuss.python.org/t/stop-allowing-deleting-things-from-pypi/17227/114>`_
+  and force-pushed over the tag history for ``python-sonarqube-api``'s source repository,
+  impeding efforts by the community to compare changes between versions
+  and obtain previous, permissively-licensed versions.
+
+* June 2024: `PySimpleGUI <https://pypi.org/project/PySimpleGUI/>`_ changed
+  licenses from LGPL to a commercial license, and deleted
+  `nearly all previous versions <https://discuss.python.org/t/48790/27>`_.
+  This resulted in widespread disruption for users, who (prior
+  to the relicensing) were downloading PySimpleGui
+  approximately 25,000 times a day. This deletion also impeded efforts
+  by the community to provide "continuity" projects, as the
+  permissively-licensed artifacts were no longer available for reference
+  and re-uploading.
+
+In addition to their disruptive effect on downstreams, deletions
+also have deleterious effects on PyPI's sustainability as well as the overall
+security of the ecosystem:
+
+* Deletions raise the baseline support load for PyPI's administrators and
+  moderators, as users mistakenly file support requests believing that PyPI
+  is broken, or that the administrators themselves have removed the
+  project.
+
+* Deletions impair incident response and external analysis, making it
+  difficult to distinguish "good faith" maintainer behavior from malicious
+  post-exploitation track-covering.
+
+The size and interdependency of the Python ecosystem is continuing to grow,
+meaning that future deletions of projects can be reasonably assumed to
+be *just as if not more* disruptive than the deletions sampled above.
+
+Given the above, this PEP concludes that the availability of "hard" deletions
+versus "soft" deletions (i.e. yanking) now presents more of a risk and detriment
+to the Python ecosystem than a benefit.
```

Comment:
This is the first mention of "soft deletions" -- perhaps just use "yank(ing)" as that's a specified term?